src/module.c

```diff
@@ -5660,16 +5683,52 @@ RedisModuleCallReply *RM_Call(RedisModuleCtx *ctx, const char *cmdname, const ch
     cmd = lookupCommand(c->argv,c->argc);
     if (!cmd) {
         errno = ENOENT;
+        if (error_as_call_replies) {
+            sds msg = sdscatfmt(sdsempty(),"ERR Unknown Redis "
+                                           "command %S.",c->argv[0]->ptr);
+            reply = callReplyCreateError(msg, ctx);
+        }
         goto cleanup;
     }
     c->cmd = c->lastcmd = c->realcmd = cmd;
 
     /* Basic arity checks. */
     if ((cmd->arity > 0 && cmd->arity != argc) || (argc < -cmd->arity)) {
         errno = EINVAL;
+        if (error_as_call_replies) {
+            sds msg = sdscatfmt(sdsempty(), "ERR Wrong number of "
+                                            "args calling Redis command %S.", c->argv[0]->ptr);
+            reply = callReplyCreateError(msg, ctx);
+        }
         goto cleanup;
     }
 
+    if (flags & REDISMODULE_ARGV_SAFE_MODE) {
```

Comment:
maybe we wanna mirror other checks currently done when scripts call redis commands, like checking `repl_good_slaves_count` and so on.
maybe we wanna do all of these just when "script mode" is enabled?

Reply:
@oranagra Added good replicas check and disk error check when script mode is enabled.